Prepublish checks: accessibility warnings

zachhale opened the pull request:
## Summary

* Adds accessibility and distribution warnings
* Adds contrast util that uses [color-contrast-checker](https://github.com/bbc/color-contrast-checker) package to help with calculating luminances and checking against WCAG guidelines

[ticket](https://app.zenhub.com/workspaces/web-stories-5e94d3aced449034e1e9b226/issues/google/web-stories-wp/4911)

## Relevant Technical Choices

* Checking test contrast against background could either be (1) against the text fill or (2) against the background image/color of elements behind it. This implementation only addresses the first scenario.
* Contrast test uses WCAG AA check (vs AAA)
* Element tappable area check chosen to be 48x48 which is what lighthouse uses

## To-do

* [x] Update to use types and enums defined in #5029 

## User-facing changes

<!-- Please describe your changes. -->

## Testing Instructions

<!-- How can the changes in this PR be verified? Relevant for QA  and user acceptance testing. -->

---

<!-- Please reference the issue(s) this PR addresses. -->

Fixes #4911

BrittanyIRL committed:
storybook for design system, starting with colors

BrittanyIRL committed:
typography for ds

BrittanyIRL committed:
typography for ds

BrittanyIRL committed:
typography for ds

BrittanyIRL committed:
typography for ds

BrittanyIRL committed:
typography for ds

BrittanyIRL committed:
WIP icons

BrittanyIRL committed:
WIP icons

BrittanyIRL committed:
WIP icons

BrittanyIRL committed:
WIP icons

BrittanyIRL committed:
WIP icons

BrittanyIRL committed:
try out subdirectories to group icons

BrittanyIRL committed:
try out subdirectories to group icons

BrittanyIRL committed:
try out subdirectories to group icons

BrittanyIRL committed:
try out subdirectories to group icons

BrittanyIRL committed:
try out subdirectories to group icons

miina committed:
Add buttons for border positioning.

miina committed:
Add display logic for outer border

miina committed:
Add outer and center border css.

miina committed:
Add border for output.

miina committed:
Add border for edit mode.

miina committed:
Nits.

BrittanyIRL committed:
organize icons like material ui

BrittanyIRL committed:
organize icons like material ui

BrittanyIRL committed:
organize icons like material ui

BrittanyIRL committed:
organize icons like material ui

BrittanyIRL committed:
organize icons like material ui

BrittanyIRL committed:
clean up sizes of svgs for dashboard custom icons

BrittanyIRL committed:
clean up size of /file svg

BrittanyIRL committed:
rest of resize to remove excess spacing on icons

BrittanyIRL committed:
everything that is one color is set to currentColor

BrittanyIRL committed:
some clean up

BrittanyIRL committed:
some clean up

BrittanyIRL committed:
some clean up

BrittanyIRL committed:
some clean up

BrittanyIRL committed:
export all icons

BrittanyIRL committed:
export all icons

BrittanyIRL committed:
export all icons

miina committed:
Rework UX for locked width.

miina committed:
Try fixing the units.

miina committed:
Adjust border for gap.

BrittanyIRL committed:
button WIP

miina committed:
Use CSS border for border.

miina committed:
Make border-width responsive for page preview.

miina committed:
Add missing propType.

miina committed:
Start adding tests for border.

miina committed:
Add border support for shape.

miina committed:
Add tests for reducer.

miina committed:
Add output tests.

BrittanyIRL committed:
clean up icon import/exports

BrittanyIRL committed:
clean up typography import/exports

BrittanyIRL committed:
clean up typography import/exports

BrittanyIRL committed:
clean up typography import/exports

BrittanyIRL committed:
clean up typography import/exports

BrittanyIRL committed:
clean up theme import/exports

BrittanyIRL committed:
continuing clean up post rebase

BrittanyIRL committed:
continuing clean up post rebase

BrittanyIRL committed:
continuing clean up post rebase

miina committed:
Remove unused code.

miina committed:
Karma update.

miina committed:
karma fix

miina committed:
Adjust video accessibility Karma.

miina committed:
Remove redundant await.

BrittanyIRL committed:
button styles and variants organized. needs colors

BrittanyIRL committed:
adding brand colors

BrittanyIRL committed:
button colors and storybook cleaned up

BrittanyIRL committed:
link as button in storybook, basic unit tests

miina committed:
Centralize checking if an element can have border.

miina committed:
PR feedback.

miina committed:
Create new icongroup component.

miina committed:
Fix Karma.

miina committed:
Fix radio navigation.

miina committed:
Use scrollIntoView.

miina committed:
Resolve conflicts.

miina committed:
Revert composer change

BrittanyIRL committed:
clean up imports

BrittanyIRL committed:
export components, clean up unnecessary styles

BrittanyIRL committed:
tab component and storybook

BrittanyIRL committed:
simplified tab group, putting functionality on parent to separate styles and functionality. some minor unit tests

BrittanyIRL committed:
missed a prop update

BrittanyIRL committed:
adding tablist role

BrittanyIRL committed:
moving typography into components

BrittanyIRL committed:
moving typography into components

BrittanyIRL committed:
moving typography into components

BrittanyIRL committed:
moving typography into components

BrittanyIRL committed:
moving typography into components

BrittanyIRL committed:
fixing rebase issues

BrittanyIRL committed:
catch tabs up to typography theme constant update

BrittanyIRL committed:
import organization on stories

BrittanyIRL committed:
removing border radius change on active round buttons

BrittanyIRL committed:
simplifying keys for tabGroups by using title and id - i had previously misread the new eslint rule about indexes as IDs. Removing redundancies for aria props

zachhale committed:
Add todos for accessibility and distribution checks

zachhale committed:
Add todos for accessibility and distribution checks

zachhale committed:
Add story excerpt checker

zachhale committed:
Add story excerpt checker

zachhale committed:
Add check for low contrast font against background color

zachhale committed:
Add check for low contrast font against background color

zachhale committed:
Add test against missing spans

zachhale committed:
Add test against missing spans

zachhale committed:
Add check for font size too small

zachhale committed:
Add check for font size too small

zachhale committed:
Add checks for low pixel density

zachhale committed:
Add checks for low pixel density

zachhale committed:
Add title, alt, and captions checks for videos

zachhale committed:
Add title, alt, and captions checks for videos

zachhale committed:
Add check for too many links on page

zachhale committed:
Add check for too many links on page

BrittanyIRL committed:
clean up theme color structure and reference material in storybook after maxs feedback. toggle light and dark mode in storybook, consistent struture for coloring object. Clean up looping

BrittanyIRL committed:
use css from styled components to get syntax highlighting and a neutral text to show size difference of typography items

BrittanyIRL committed:
catch up

swissspidy committed:
3P Media: Add poster image for GIF video elements

Also prevents static poster image from being used in the editor, making the GIF appear broken

swissspidy committed:
Add more tests

zachhale committed:
Add image missing alt check

zachhale committed:
Add image missing alt check

BrittanyIRL committed:
gut unnecessary tests

BrittanyIRL committed:
looping clean up

zachhale committed:
Add tappable region check

zachhale committed:
Add tappable region check

zachhale committed:
Comment tweak

zachhale committed:
Comment tweak

zachhale committed:
Refactor out contrast utils

zachhale committed:
Refactor out contrast utils

zachhale committed:
Allow resource to have attributes

zachhale committed:
Allow resource to have attributes

carloskelly13 committed:
Added base animation effect chooser

swissspidy committed:
Fix PHP-CSS-Parser patching

BrittanyIRL committed:
adding modal component to design system, works the same as exsiting modal component in edit story and dashboard except theme is passed in so that the background color of the overlay can be respectful of light or dark theme. defaults to dark hex.

BrittanyIRL committed:
wip

BrittanyIRL committed:
swap to border box

BrittanyIRL committed:
wip

BrittanyIRL committed:
ensure centering when text is shorter than button width min

embarks reviewed (commented):
So I'm working on [this checklist for errors](https://app.zenhub.com/workspaces/web-stories-5e94d3aced449034e1e9b226/issues/google/web-stories-wp/4910) and I can see this one is for warnings. The other reviews are for guidance. I organized my "guidance"-severity level modules `media` and `general` in dirs named after their severity level, should we replicate that pattern for "warnings" and "erros"? Is it a useful pattern? 

I was thinking an index file for the severity like `guidance/index.js` or `warnings/index.js` _might_ be a good way to abstract some of the "controlling" logic (such as, organizing element-, page-, and story-level checkers) was my thought. Or should we just go with `prepublish/*` top-level modules for now?

BrittanyIRL committed:
dialog and modal clean up in ds

BrittanyIRL committed:
swap demo cancel button to tertiary to be in line with figma visually instead of based on label

BrittanyIRL committed:
some clean up

Commit:
prepublish/errors/metadata: check for critical metadata, set up tests and todos

carloskelly13 committed:
Finished design and added most animations.

dependabot-preview[bot] committed:
Bump civicrm/composer-downloads-plugin from 2.1.1 to 3.0.1

Bumps [civicrm/composer-downloads-plugin](https://github.com/civicrm/composer-downloads-plugin) from 2.1.1 to 3.0.1.
- [Release notes](https://github.com/civicrm/composer-downloads-plugin/releases)
- [Commits](https://github.com/civicrm/composer-downloads-plugin/compare/v2.1.1...v3.0.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
Bump php-stubs/wordpress-stubs from 5.5.1 to 5.5.3

Bumps [php-stubs/wordpress-stubs](https://github.com/php-stubs/wordpress-stubs) from 5.5.1 to 5.5.3.
- [Release notes](https://github.com/php-stubs/wordpress-stubs/releases)
- [Commits](https://github.com/php-stubs/wordpress-stubs/compare/v5.5.1...v5.5.3)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

swissspidy committed:
Install composer-normalize via phar

composer-normalize requires PHP 7.2+ and Composer 2.
This way of installing it prevents any version conflicts.

dependabot-preview[bot] committed:
Bump @babel/core from 7.11.6 to 7.12.3

Bumps [@babel/core](https://github.com/babel/babel/tree/HEAD/packages/babel-core) from 7.11.6 to 7.12.3.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.12.3/packages/babel-core)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
Bump @wordpress/data from 4.23.1 to 4.25.0

Bumps [@wordpress/data](https://github.com/WordPress/gutenberg/tree/HEAD/packages/data) from 4.23.1 to 4.25.0.
- [Release notes](https://github.com/WordPress/gutenberg/releases)
- [Changelog](https://github.com/WordPress/gutenberg/blob/master/changelog.txt)
- [Commits](https://github.com/WordPress/gutenberg/commits/@wordpress/data@4.25.0/packages/data)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
Bump eslint-plugin-react from 7.21.3 to 7.21.5

Bumps [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react) from 7.21.3 to 7.21.5.
- [Release notes](https://github.com/yannickcr/eslint-plugin-react/releases)
- [Changelog](https://github.com/yannickcr/eslint-plugin-react/blob/master/CHANGELOG.md)
- [Commits](https://github.com/yannickcr/eslint-plugin-react/compare/v7.21.3...v7.21.5)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

swissspidy committed:
Use remote patches

swissspidy committed:
Update CI config

swissspidy committed:
Update composer config

Remove dupe

carloskelly13 committed:
Update container to buttons

BrittanyIRL committed:
catch up from rebase

BrittanyIRL committed:
Merge pull request #4893 from google/feature/ds-storybook-foundation

Design System Storybook (new theme colors)

BrittanyIRL committed:
Merge pull request #4894 from google/feature/ds-storybook-typography

Design System Storybook - Typography

zachhale committed:
Merge branch 'main' into feature/4911-prepublish-accessibility-warnings

Commit:
prepublish/errors/metadata: tests and logic/obj shape fixes

carloskelly13 committed:
Switch to pure CSS solution

BrittanyIRL committed:
clean up from rebase, helper to unpack theme typography presets without typing it every time, make storybook easier to reference

BrittanyIRL committed:
Update assets/src/design-system/components/tabGroup/index.js

Co-authored-by: Max Yinger <35983235+littlemilkstudio@users.noreply.github.com>

BrittanyIRL committed:
clean up rebase

carloskelly13 committed:
Finished 6 animations.

BrittanyIRL committed:
helper for box shadow css in theme, move font preset to theme/helpers. give tab and tabGroup a dark theme background since tabs are only for editor and the coloring is kind of confusing in storybook.

BrittanyIRL committed:
rebase keeps bringing these unnecessary tests back from the dead

swissspidy committed:
Merge pull request #5090 from google/dependabot/composer/civicrm/composer-downloads-plugin-3.0.1

Bump civicrm/composer-downloads-plugin from 2.1.1 to 3.0.1

swissspidy committed:
Merge pull request #5091 from google/dependabot/composer/php-stubs/wordpress-stubs-5.5.3

Bump php-stubs/wordpress-stubs from 5.5.1 to 5.5.3

swissspidy committed:
Merge pull request #5095 from google/dependabot/npm_and_yarn/babel/core-7.12.3

Bump @babel/core from 7.11.6 to 7.12.3

dependabot-preview[bot] committed:
Bump @testing-library/react from 11.0.4 to 11.1.0 (#5099)

swissspidy committed:
Merge pull request #5096 from google/dependabot/npm_and_yarn/wordpress/data-4.25.0

Bump @wordpress/data from 4.23.1 to 4.25.0

dependabot-preview[bot] committed:
Bump polished from 3.6.7 to 4.0.3 (#5094)

swissspidy committed:
Merge pull request #5100 from google/dependabot/npm_and_yarn/eslint-plugin-react-7.21.5

Bump eslint-plugin-react from 7.21.3 to 7.21.5

swissspidy committed:
Merge branch 'main' into fix/composer-patches

Commit:
remove old comments

BrittanyIRL committed:
hoist regex out of map

BrittanyIRL committed:
Merge pull request #5003 from google/feature/ds-storybook-buttons

Design System Storybook - Buttons

BrittanyIRL committed:
Merge pull request #5007 from google/feature/ds-storybook-floating-tab

Design System Storybook - Tabs

swissspidy committed:
Merge branch 'main' into fix/composer-normalize

BrittanyIRL committed:
catch up from rebase

BrittanyIRL committed:
remove unnecessary transition

BrittanyIRL committed:
more cleanup

BrittanyIRL committed:
update overlay color per sam

swissspidy committed:
Dashboard: do not send body when deleting

swissspidy committed:
Move all REST API endpoints under common namespace

swissspidy committed:
Use `_method=DELETE` instead for deletion

swissspidy committed:
Update tests

zachhale committed:
Update prepublish checks to new filetructure and types

zachhale committed:
Update prepublish checks to new filetructure and types

swissspidy committed:
Also update `deleteMedia` in editor accordingly

BrittanyIRL committed:
re-add background color

carloskelly13 committed:
Fix import order.

swissspidy committed:
Update dashboard preloading

swissspidy committed:
Ignore new classes from coverage

miina committed:
Merge pull request #4895 from google/add/1051-border-panel

Add border panel

carloskelly13 committed:
Merge pull request #5110 from google/feature/animation-effect-chooser

Animations: Effect Chooser Component

swissspidy committed:
Merge pull request #5082 from google/fix/composer-patches

swissspidy committed:
Merge pull request #5055 from google/fix/gif-poster

swissspidy committed:
Merge branch 'main' into fix/composer-normalize

Commit:
prepublish/errors/metadata(fix): page id for page attachments

zachhale committed:
Use null propogation ?

zachhale committed:
Use null propogation ?

swissspidy committed:
Merge pull request #5120 from google/try/sucuri

maxyinger committed:
Merge pull request #4906 from google/3805-animation-parts-as-limited-effects

Animations: Animation Parts Seen as a Limited Effect to user

Commit:
[docs] Update the getting-started documentation, includes the composer setup.

Commit:
Merge pull request #5086 from google/inc/prepublish/4910/missing-critical-metadata

prepublish checklist - missing critical metadata

swissspidy committed:
Merge pull request #5092 from google/fix/composer-normalize

swissspidy committed:
Travis: force Composer v1.x

BrittanyIRL committed:
Merge pull request #5085 from google/feature/ds-dialog-4956

Design System Storybook - Dialog (and a base modal)

swissspidy committed:
Merge pull request #5144 from google/zlyu

Update Getting Started documentation

dependabot-preview[bot] committed:
Bump react-color from 2.18.1 to 2.19.3

Bumps [react-color](https://github.com/casesandberg/react-color) from 2.18.1 to 2.19.3.
- [Release notes](https://github.com/casesandberg/react-color/releases)
- [Commits](https://github.com/casesandberg/react-color/commits)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
Bump @wordpress/e2e-test-utils from 4.14.0 to 4.15.0

Bumps [@wordpress/e2e-test-utils](https://github.com/WordPress/gutenberg/tree/HEAD/packages/e2e-test-utils) from 4.14.0 to 4.15.0.
- [Release notes](https://github.com/WordPress/gutenberg/releases)
- [Changelog](https://github.com/WordPress/gutenberg/blob/master/changelog.txt)
- [Commits](https://github.com/WordPress/gutenberg/commits/@wordpress/e2e-test-utils@4.15.0/packages/e2e-test-utils)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
Bump eslint-plugin-jest-dom from 3.2.3 to 3.2.4

Bumps [eslint-plugin-jest-dom](https://github.com/testing-library/eslint-plugin-jest-dom) from 3.2.3 to 3.2.4.
- [Release notes](https://github.com/testing-library/eslint-plugin-jest-dom/releases)
- [Commits](https://github.com/testing-library/eslint-plugin-jest-dom/compare/v3.2.3...v3.2.4)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
Bump css-loader from 4.3.0 to 5.0.1

Bumps [css-loader](https://github.com/webpack-contrib/css-loader) from 4.3.0 to 5.0.1.
- [Release notes](https://github.com/webpack-contrib/css-loader/releases)
- [Changelog](https://github.com/webpack-contrib/css-loader/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack-contrib/css-loader/compare/v4.3.0...v5.0.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
Bump @babel/eslint-plugin from 7.11.5 to 7.12.1

Bumps [@babel/eslint-plugin](https://github.com/babel/babel/tree/HEAD/eslint/babel-eslint-plugin) from 7.11.5 to 7.12.1.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.12.1/eslint/babel-eslint-plugin)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

swissspidy committed:
Merge pull request #5156 from google/dependabot/npm_and_yarn/wordpress/e2e-test-utils-4.15.0

Bump @wordpress/e2e-test-utils from 4.14.0 to 4.15.0

swissspidy committed:
Merge pull request #5157 from google/dependabot/npm_and_yarn/eslint-plugin-jest-dom-3.2.4

Bump eslint-plugin-jest-dom from 3.2.3 to 3.2.4

swissspidy committed:
Update list of Google Fonts

swissspidy committed:
Merge pull request #5162 from google/dependabot/npm_and_yarn/babel/eslint-plugin-7.12.1

Bump @babel/eslint-plugin from 7.11.5 to 7.12.1

swissspidy committed:
Merge pull request #5155 from google/dependabot/npm_and_yarn/react-color-2.19.3

Bump react-color from 2.18.1 to 2.19.3

swissspidy committed:
Merge pull request #5166 from google/update/google-fonts

Update list of Google Fonts

swissspidy committed:
Merge pull request #5161 from google/dependabot/npm_and_yarn/css-loader-5.0.1

Bump css-loader from 4.3.0 to 5.0.1

dependabot-preview[bot] committed:
Create Dependabot config file

dependabot-preview[bot] committed:
Bump core-js from 3.6.5 to 3.7.0

Bumps [core-js](https://github.com/zloirock/core-js) from 3.6.5 to 3.7.0.
- [Release notes](https://github.com/zloirock/core-js/releases)
- [Changelog](https://github.com/zloirock/core-js/blob/master/CHANGELOG.md)
- [Commits](https://github.com/zloirock/core-js/compare/v3.6.5...v3.7.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

swissspidy committed:
Weekly intervals

dependabot-preview[bot] committed:
Bump eslint from 7.10.0 to 7.13.0

Bumps [eslint](https://github.com/eslint/eslint) from 7.10.0 to 7.13.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.10.0...v7.13.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
Bump webpack-cli from 4.0.0 to 4.2.0

Bumps [webpack-cli](https://github.com/webpack/webpack-cli) from 4.0.0 to 4.2.0.
- [Release notes](https://github.com/webpack/webpack-cli/releases)
- [Changelog](https://github.com/webpack/webpack-cli/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-cli/compare/webpack-cli@4.0.0...webpack-cli@4.2.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
Bump got from 11.7.0 to 11.8.0

Bumps [got](https://github.com/sindresorhus/got) from 11.7.0 to 11.8.0.
- [Release notes](https://github.com/sindresorhus/got/releases)
- [Commits](https://github.com/sindresorhus/got/compare/v11.7.0...v11.8.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
Bump big.js from 6.0.1 to 6.0.2 (#5172)

dependabot-preview[bot] committed:
Bump use-context-selector from 1.1.4 to 1.2.11 (#5170)

dependabot-preview[bot] committed:
Bump styled-components from 5.1.1 to 5.2.1

Bumps [styled-components](https://github.com/styled-components/styled-components) from 5.1.1 to 5.2.1.
- [Release notes](https://github.com/styled-components/styled-components/releases)
- [Changelog](https://github.com/styled-components/styled-components/blob/v5.2.1/CHANGELOG.md)
- [Commits](https://github.com/styled-components/styled-components/compare/v5.1.1...v5.2.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

swissspidy committed:
Update eslint patch file

swissspidy committed:
Merge pull request #5168 from google/dependabot/add-v2-config-file

dependabot-preview[bot] committed:
Bump thread-loader from 3.0.0 to 3.0.1 (#5174)

dependabot[bot] committed:
Bump actions/setup-node from v1 to v2.1.2

Bumps [actions/setup-node](https://github.com/actions/setup-node) from v1 to v2.1.2.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v1...c6fd00ceb9747fb23ffdf72987450a2664414867)

Signed-off-by: dependabot[bot] <support@github.com>

swissspidy committed:
Merge pull request #5171 from google/dependabot/npm_and_yarn/webpack-cli-4.2.0

Bump webpack-cli from 4.0.0 to 4.2.0

dependabot[bot] committed:
Bump @babel/eslint-parser from 7.11.5 to 7.12.1

Bumps [@babel/eslint-parser](https://github.com/babel/babel/tree/HEAD/eslint/babel-eslint-parser) from 7.11.5 to 7.12.1.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.12.1/eslint/babel-eslint-parser)

Signed-off-by: dependabot[bot] <support@github.com>

swissspidy committed:
Merge pull request #5176 from google/dependabot/npm_and_yarn/styled-components-5.2.1

Bump styled-components from 5.1.1 to 5.2.1

dependabot[bot] committed:
Bump thread-loader from 3.0.0 to 3.0.1

Bumps [thread-loader](https://github.com/webpack-contrib/thread-loader) from 3.0.0 to 3.0.1.
- [Release notes](https://github.com/webpack-contrib/thread-loader/releases)
- [Changelog](https://github.com/webpack-contrib/thread-loader/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack-contrib/thread-loader/compare/v3.0.0...v3.0.1)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump source-map-loader from 1.1.0 to 1.1.2

Bumps [source-map-loader](https://github.com/webpack-contrib/source-map-loader) from 1.1.0 to 1.1.2.
- [Release notes](https://github.com/webpack-contrib/source-map-loader/releases)
- [Changelog](https://github.com/webpack-contrib/source-map-loader/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack-contrib/source-map-loader/compare/v1.1.0...v1.1.2)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump jest-axe from 4.0.0 to 4.1.0

Bumps [jest-axe](https://github.com/nickcolley/jest-axe) from 4.0.0 to 4.1.0.
- [Release notes](https://github.com/nickcolley/jest-axe/releases)
- [Changelog](https://github.com/nickcolley/jest-axe/blob/main/CHANGELOG.md)
- [Commits](https://github.com/nickcolley/jest-axe/compare/v4.0.0...v4.1.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump webpack-cli from 4.0.0 to 4.2.0

Bumps [webpack-cli](https://github.com/webpack/webpack-cli) from 4.0.0 to 4.2.0.
- [Release notes](https://github.com/webpack/webpack-cli/releases)
- [Changelog](https://github.com/webpack/webpack-cli/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-cli/compare/webpack-cli@4.0.0...webpack-cli@4.2.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump @babel/preset-env from 7.11.5 to 7.12.1

Bumps [@babel/preset-env](https://github.com/babel/babel/tree/HEAD/packages/babel-preset-env) from 7.11.5 to 7.12.1.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.12.1/packages/babel-preset-env)

Signed-off-by: dependabot[bot] <support@github.com>

swissspidy committed:
Merge pull request #5169 from google/dependabot/npm_and_yarn/eslint-7.13.0

swissspidy committed:
Merge pull request #5160 from google/dependabot/npm_and_yarn/core-js-3.7.0

Bump core-js from 3.6.5 to 3.7.0

swissspidy committed:
Merge pull request #5175 from google/dependabot/npm_and_yarn/got-11.8.0

Bump got from 11.7.0 to 11.8.0

swissspidy committed:
Merge pull request #5177 from google/dependabot/github_actions/actions/setup-node-v2.1.2

Bump actions/setup-node from v1 to v2.1.2

swissspidy committed:
Merge pull request #5179 from google/dependabot/npm_and_yarn/babel/eslint-parser-7.12.1

Bump @babel/eslint-parser from 7.11.5 to 7.12.1

dependabot[bot] committed:
Bump eslint-plugin-react-hooks from 4.1.2 to 4.2.0

Bumps [eslint-plugin-react-hooks](https://github.com/facebook/react/tree/HEAD/packages/eslint-plugin-react-hooks) from 4.1.2 to 4.2.0.
- [Release notes](https://github.com/facebook/react/releases)
- [Changelog](https://github.com/facebook/react/blob/master/packages/eslint-plugin-react-hooks/CHANGELOG.md)
- [Commits](https://github.com/facebook/react/commits/HEAD/packages/eslint-plugin-react-hooks)

Signed-off-by: dependabot[bot] <support@github.com>

swissspidy committed:
Merge pull request #5184 from google/dependabot/npm_and_yarn/source-map-loader-1.1.2

Bump source-map-loader from 1.1.0 to 1.1.2

swissspidy committed:
Merge pull request #5182 from google/dependabot/npm_and_yarn/thread-loader-3.0.1

Bump thread-loader from 3.0.0 to 3.0.1

swissspidy committed:
Merge pull request #5187 from google/dependabot/npm_and_yarn/webpack-cli-4.2.0

Bump webpack-cli from 4.0.0 to 4.2.0

swissspidy committed:
Merge pull request #5188 from google/dependabot/npm_and_yarn/babel/preset-env-7.12.1

Bump @babel/preset-env from 7.11.5 to 7.12.1

swissspidy committed:
Merge pull request #5185 from google/dependabot/npm_and_yarn/jest-axe-4.1.0

Bump jest-axe from 4.0.0 to 4.1.0

swissspidy committed:
Update patch file

swissspidy committed:
Merge pull request #5181 from google/dependabot/npm_and_yarn/eslint-plugin-react-hooks-4.2.0

zachhale committed:
Add prepublish text length guidelines (#5033)

* Add prepublish text length guidelines

* Switch from warning to guidance

* Add i18n to text guidances

* Add typedefs

* Minor language fixes

* Migrate to using new constants and types

zachhale committed:
Merge branch 'feature/4911-prepublish-accessibility-warnings' of github.com:google/web-stories-wp into feature/4911-prepublish-accessibility-warnings
